#! /bin/sh /usr/share/dpatch/dpatch-run
## 22_avahi_client_0.6_api.dpatch by Daniel T Chen <crimsun@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Update from svn changeset 13520 to enable Avahi client 0.6 API support

@DPATCH@

diff -uNr vlc-0.8.4.debian/modules/access_output/bonjour.c vlc-0.8.4.debian.work/modules/access_output/bonjour.c
--- vlc-0.8.4.debian/modules/access_output/bonjour.c	2005-11-26 06:37:47.000000000 -0800
+++ vlc-0.8.4.debian.work/modules/access_output/bonjour.c	2006-03-05 18:29:38.000000000 -0800
@@ -33,6 +33,8 @@
 #include <vlc/sout.h>
 
 #include <avahi-client/client.h>
+#include <avahi-client/publish.h>
+#include <avahi-client/lookup.h>
 #include <avahi-common/alternative.h>
 #include <avahi-common/simple-watch.h>
 #include <avahi-common/malloc.h>
@@ -114,7 +116,7 @@
     }
 
     error = avahi_entry_group_add_service( p_sys->group, AVAHI_IF_UNSPEC,
-                                           AVAHI_PROTO_UNSPEC, p_sys->psz_name,
+                                           AVAHI_PROTO_UNSPEC, 0, p_sys->psz_name,
                                            p_sys->psz_stype, NULL, NULL,
                                            p_sys->i_port,
                                            p_sys->psz_txt, NULL );
@@ -155,7 +157,8 @@
         if( p_sys->group != NULL )
             avahi_entry_group_reset( p_sys->group );
     }
-    else if( state == AVAHI_CLIENT_DISCONNECTED )
+    else if( state == AVAHI_CLIENT_FAILURE &&
+              (avahi_client_errno(c) == AVAHI_ERR_DISCONNECTED) )
     {
         msg_Err( p_sys->p_log, "avahi client disconnected" );
         avahi_simple_poll_quit( p_sys->simple_poll );
@@ -221,6 +224,7 @@
     }
 
     p_sys->client = avahi_client_new( avahi_simple_poll_get(p_sys->simple_poll),
+                                      0,
                                       client_callback, p_sys, &err );
     if( p_sys->client == NULL )
     {
diff -uNr vlc-0.8.4.debian/modules/services_discovery/bonjour.c vlc-0.8.4.debian.work/modules/services_discovery/bonjour.c
--- vlc-0.8.4.debian/modules/services_discovery/bonjour.c	2005-11-26 06:37:50.000000000 -0800
+++ vlc-0.8.4.debian.work/modules/services_discovery/bonjour.c	2006-03-05 18:30:37.000000000 -0800
@@ -30,6 +30,8 @@
 #include <vlc/intf.h>
 
 #include <avahi-client/client.h>
+#include <avahi-client/publish.h>
+#include <avahi-client/lookup.h>
 #include <avahi-common/simple-watch.h>
 #include <avahi-common/malloc.h>
 #include <avahi-common/error.h>
@@ -82,7 +84,8 @@
     services_discovery_t *p_sd = ( services_discovery_t* )userdata;
     services_discovery_sys_t *p_sys = p_sd->p_sys;
 
-    if( state == AVAHI_CLIENT_DISCONNECTED )
+    if( state == AVAHI_CLIENT_FAILURE &&
+        (avahi_client_errno(c) == AVAHI_ERR_DISCONNECTED) )
     {
         msg_Err( p_sd, "avahi client disconnected" );
         avahi_simple_poll_quit( p_sys->simple_poll );
@@ -104,12 +107,13 @@
     const AvahiAddress *address,
     uint16_t port,
     AvahiStringList *txt,
+    AvahiLookupResultFlags flags,
     void* userdata )
 {
     services_discovery_t *p_sd = ( services_discovery_t* )userdata;
     services_discovery_sys_t *p_sys = p_sd->p_sys;
 
-    if( event == AVAHI_RESOLVER_TIMEOUT )
+    if( event == AVAHI_RESOLVER_FAILURE )
     {
         msg_Err( p_sd,
                  "failed to resolve service '%s' of type '%s' in domain '%s'",
@@ -186,6 +190,7 @@
     const char *name,
     const char *type,
     const char *domain,
+    AvahiLookupResultFlags flags,
     void* userdata )
 {
     services_discovery_t *p_sd = ( services_discovery_t* )userdata;
@@ -195,6 +200,7 @@
     {
         if( avahi_service_resolver_new( p_sys->client, interface, protocol,
                                         name, type, domain, AVAHI_PROTO_UNSPEC,
+                                        0,
                                         resolve_callback, userdata ) == NULL )
         {
             msg_Err( p_sd, "failed to resolve service '%s': %s", name,
@@ -246,6 +252,7 @@
     }
 
     p_sys->client = avahi_client_new( avahi_simple_poll_get(p_sys->simple_poll),
+                                      0,
                                       client_callback, p_sd, &err );
     if( p_sys->client == NULL )
     {
@@ -257,6 +264,7 @@
     p_sys->sb = avahi_service_browser_new( p_sys->client, AVAHI_IF_UNSPEC,
                                            AVAHI_PROTO_UNSPEC,
                                            "_vlc-http._tcp", NULL,
+                                           0,
                                            browse_callback, p_sd );
     if( p_sys->sb == NULL )
     {
